shared/fs/filepreview/index.js

```diff
@@ -0,0 +1,104 @@
+// @flow
+import * as React from 'react'
+import * as Types from '../../constants/types/fs'
+import {globalStyles, globalColors, globalMargins, isMobile} from '../../styles'
+import {Box, BackButton, Button, ClickableBox, Icon, List, Text} from '../../common-adapters'
+import {type IconType} from '../../common-adapters/icon'
+import {pathToIcon} from '../icon'
+import {formatTimeForMessages} from '../../util/timestamp'
+
+
+const stylesCommonCore = {
+  alignItems: 'center',
+  borderBottomColor: globalColors.black_05,
+  borderBottomWidth: 1,
+  justifyContent: 'center',
+}
+
+const stylesCommonColumn = {
+  ...globalStyles.flexBoxColumn,
+  ...stylesCommonCore,
+  minHeight: isMobile ? 64 : 40,
+}
+
+type FilePreviewHeaderProps = {
+  title: string,
+  desc: string,
+}
+
+const filePreviewHeaderStyle = {...stylesCommonColumn, alignItems: 'center', borderBottomWidth: 0}
+
+const FilePreviewHeader = ({title,desc}: FilePreviewHeaderProps) => (
+  <Box>
+    <Box style={filePreviewHeaderStyle}>
+      <Text type="BodyBig">
+      {title}
+      </Text>
+      <Text type="BodySmall">{desc}</Text>
+    </Box>
+  </Box>
+)
+
+type FilePreviewProps = {
+  path: Types.Path,
+  meta: Types.PathItemMetadata,
+  onBack: () => void,
+}
+
+const stylesContainer = {
+  ...globalStyles.flexBoxColumn,
+  ...globalStyles.fullHeight,
+  flex: 1,
+  alignItems: 'center',
+  justifyContent: 'center',
+}
+const styleOuterContainer = {
+  height: '100%',
+  position: 'relative',
+}
+
+const humanReadableFileSize = (size) => {
+  if(size > 1024*1024*1024*1024)
+    return ''+Math.round(size/(1024*1024*1024*1024))+'tb'
+  if(size > 1024*1024*1024)
+    return ''+Math.round(size/(1024*1024*1024))+'gb'
+  if(size > 1024*1024)
+    return ''+Math.round(size/(1024*1024))+'mb'
+  if(size > 1024)
+    return ''+Math.round(size/1024)+'kb'
+  return ''+size
+}
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  onBack: () => dispatch(navigateUp()),
+})
+
+class FilePreview extends React.PureComponent<FilePreviewProps> {
```

Comment:
I know this is a WIP, it seems like its partially connected / partially not?
I'd advocate moving most of the logic into the container and make the dumb component 'dumber'.
So things like the string building (humanReadableFileSize) would be done in the mergeProps, etc